app/services/govspeak_document/in_app_options.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class GovspeakDocument::InAppOptions < GovspeakDocument::Options
+  include Rails.application.routes.url_helpers
+
+  def to_h
+    super.to_h.merge(images: in_app_images)
+  end
+
+private
+
+  def in_app_images
+    edition.revision.image_revisions.map do |image_revision|
+      image_attributes(image_revision).merge(
+        # This is the same as url_for(image_revision.crop_variant) in the view, or
+        # image_revision.crop_variant.service_url, except that these don't work outside
+        # of a request context which normally includes the request host.
+        url: rails_representation_path(image_revision.crop_variant, only_path: true),
+      )
+    end
+  end
+
+  def image_attributes(image_revision)
```

Comment:
Since the data here isn't complete until we merge in a url parameter I think this method ends up being a bit misleadingly named.

I'd suggest though that you probably don't need it to be a separate method and could just have the whole hash in the in_app_images method to save having to context switch when reading the code.

Reply:
Agreed - done.